core/controllers/contributor_dashboard.py

```diff
@@ -363,48 +362,20 @@ def _get_reviewable_exploration_opportunity_summaries(
         Raises:
             Exception. No exploration_id found for the node_id.
         """
-        # 1. Fetch the eligible topics.
-        # 2. Fetch the stories for the topics.
-        # 3. Get the reviewable translation suggestion target IDs for the user.
-        # 4. Get story exploration nodes in order, filtering for explorations
-        # that have in review translation suggestions.
-        if topic_name is None:
-            topics = topic_fetchers.get_all_topics()
-        else:
-            topic = topic_fetchers.get_topic_by_name(topic_name)
-            if topic is None:
-                raise self.InvalidInputException(
-                    'The supplied input topic: %s is not valid' % topic_name)
-            topics = [topic]
-        topic_stories = story_fetchers.get_stories_by_ids(
-            [
-                reference.story_id
-                for topic in topics
-                for reference in topic.get_all_story_references()
-                if reference.story_is_published
-            ],
-            strict=True
-        )
-        topic_exp_ids = []
-        for story in topic_stories:
-            for node in story.story_contents.get_ordered_nodes():
-                if node.exploration_id is None:
-                    raise Exception(
-                        'No exploration_id found for the node_id: %s'
-                        % node.id
-                    )
-                topic_exp_ids.append(node.exploration_id)
```

Comment:
Why do we need to remove these steps?

Reply:
@chris7716 Because these steps fetch the exploration IDs belonging to the topics' stories and require multiple calls to the datastore. The changes my PR makes allows access to the stories' exploration IDs from the corresponding `TopicModel`, making it quicker to do the same thing, while delegating the collecting of a topic's stories' exploration IDs to the service layer.